### PR TITLE
Code Updates to Arrangement Plugins, Attribute Editor and Conversation View 

### PR DESCRIPTION
**Pull request by Delphinus8821**
### Prerequisites

- [x] Reviewed the [checklist](CHECKLIST.md)

- [ ] Reviewed feedback from the "Sonar Cloud" bot. Note that you have to wait
    for the "CI / Unit Tests") to complete first. Failed Unit tests can be 
    debugged by adding the label "verbose logging" to the GitHub PR.

### Description of the Change
Updated Core Arrangement Plugins, Core Attribute Editor and Core Conversation View with code features from the new Java versions and some general cleanup.

### Why Should This Be In Core?
Updates to Core code.

### Applicable Issues
#1987

### CODE REVIEW

File: CoreArrangementPlugins/src/au/gov/asd/tac/constellation/plugins/arrangements/gather/GatherNodesContextMenu.java [hunk 1]
```diff
@@ -64,18 +64,10 @@ public List<String> getItems(final GraphReadMethods graph, final GraphElementTyp
...
-                break;
-            case VERTEX:
+            case VERTEX ->
                 PluginExecution.withPlugin(new GatherNodesForVertexContextMenuPlugin(element)).executeLater(graph);
-                break;
-            case META:
-            case LINK:
-            case EDGE:
-            case TRANSACTION:
-            default:
```
Comment (antares1470) [inline CoreArrangementPlugins/src/au/gov/asd/tac/constellation/plugins/arrangements/gather/GatherNodesContextMenu.java:77]: do we need a default case added here?

File: CoreArrangementPlugins/src/au/gov/asd/tac/constellation/plugins/arrangements/planar/PQNode.java [hunk 2]
```diff
@@ -332,9 +332,7 @@ public List<PQNode> cleanSinglyPartialQNode() {
                         carry += grandchild.numLeafDescendants;
                         reverseCount += grandchild.numLeafDescendants;
                     }
-                    break;
-                default:
```
Comment (antares1470) [inline CoreArrangementPlugins/src/au/gov/asd/tac/constellation/plugins/arrangements/planar/PQNode.java:336]: same here

File: CoreArrangementPlugins/src/au/gov/asd/tac/constellation/plugins/arrangements/planar/PQNode.java [hunk 3]
```diff
@@ -393,9 +389,7 @@ public List<PQNode> cleanDoublyPartialQNode() {
                             count += grandchild.numLeafDescendants;
                         }
                     }
-                    break;
-                default:
```
Comment (antares1470) [inline CoreArrangementPlugins/src/au/gov/asd/tac/constellation/plugins/arrangements/planar/PQNode.java:397]: and here

File: CoreArrangementPlugins/src/au/gov/asd/tac/constellation/plugins/arrangements/planar/PQTree.java [hunk 4]
```diff
@@ -184,25 +184,23 @@ public int readAndRemoveDirectionIndicator() {
...
                 }
-                break;
-            case LEAF_NODE:
+            }
+            case LEAF_NODE -> {
                 if (node.getVirtualNum() == currentNumber) {
                     frontier.add(node.getRealNum());
                 }
-                break;
-            default:
```
Comment (antares1470) [inline CoreArrangementPlugins/src/au/gov/asd/tac/constellation/plugins/arrangements/planar/PQTree.java:204]: and here

File: CoreArrangementPlugins/src/au/gov/asd/tac/constellation/plugins/arrangements/time/LayerByTimePlugin.java [hunk 5]
```diff
@@ -669,22 +669,20 @@ private int getDuplicateNode(final GraphWriteMethods graph, final Map<String, In
...
-            case VERTEX:
+            }
+            case VERTEX -> {
                 for (int i = 0; i < graph.getAttributeCount(GraphElementType.VERTEX); i++) {
                     final int attr = graph.getAttribute(GraphElementType.VERTEX, i);
                     final Object value = graph.getObjectValue(attr, fromId);
                     graph.setObjectValue(attr, toId, value);
                 }   
-                break;
-            default:
```
Comment (antares1470) [inline CoreArrangementPlugins/src/au/gov/asd/tac/constellation/plugins/arrangements/time/LayerByTimePlugin.java:686]: and here

**Review by antares1470 — APPROVED** [review]

**Review by Quasar985 — APPROVED** [review]

**Pull request merged**